### PR TITLE
VACMS-7019 - Create Promo Banner content type

### PR DESCRIPTION
**Pull request by msbtterswrth**
Create a new content type for promo banners

Relates to #7019 .

## Testing done
Reviewed new content type and saw that it fulfilled ACs from parent ticket

## Screenshots
image

## QA steps

As a cms editor
- [x] 1. Visit Full Width Alert (/node/add/banner) and confirm that the name of the content type has been changed to 'Full Width Alert ' from "Banner".
- [x] 2. Visit promo banner creation screen (node/add/promo_banner).
- [x] 3. Confirm the promo type defaults to 'announcement'.
- [x] 4. Confirm that the promo text field is required and has a character limit of 55 characters.
- [x] 5. Confirm that the url field is required and can link to internal or external urls.
- [?] 6. Confirm it utilizes same paths field and workflow as Full Width Alert 
6. Then validate Acceptance Criteria from issue
- [ ]  Content model designed and implemented
-- [ ]  Sharing as many patterns as possible with Banner content type
- [x]  Consider renaming "Banner" to "Full width banner"
- [ ]  Content model documentation has been created, ideally with a screenshot of the design and some annotation for field names. (possible candidate for some Confluence POC for content model documentation)

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [] `Platform CMS team`
- [x] `Sitewide CMS Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [x] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by kevwalsh — CHANGES_REQUESTED** [review]
Nice!

All super minor. 

- [ ] Needs a content type description. [From the design system documentation](https://design.va.gov/components/promo-banners), let's get go with "Promo banners are fixed content used for dismissible announcements such as new tools, news, etc."

 <details><summary>Screenshot</summary>

Add_content___VA_gov_CMS


</details>

Some light (MVP) editorial experience work

- [ ] 1. We should explain what Title is for. It could either be a description of promo banner (Title label could be "Name of promo banner", description could be "A short description of its contents") OR we could use make the node title the "Promo text" field, which is the approach used for the Full Width Alert content type, where the node title is the "Heading" field, and the placed lower on the form than the select list.  I think this is the only semi-substantive editorial experience design decision that we are making right now. To me, less is more, so i think the Title field should be come the Link text field. 
- [ ] 2. "www.va.gov supports three types of promo banners" not "Currently, the CMS supports". Ideally, we should fix that on the Full Width Alert type too. 
- [ ] 3. "promo types" not "alert types"
- [ ] 4. I think "Link text" is more consistent with other link fields, vs "Promo text". OR, we somehow indicate with help text that this text will be the link text. But less is more. 
- [ ] 5. This field needs to be set to "55 characters remaining." using the textcounter field widget formatter. We have an established syntax/pattern for this
- [ ] 6. Ugh that drupal default URL help text. Easiest solution is just to switch to Linkit. 

<details><summary>Screenshot</summary>

Cursor_and_Create_Promo_Banner___VA_gov_CMS


</details>


A few other things, might as well take care of them now

- [ ] Needs pathauto. how about "/promo-banner/[title]".  Eventually we may want to move a bunch of "block" type content into something like "/block/".
- [ ] Hide two fields on the node display: the editorial workflow widget, and the Section field. 

<details><summary>Screenshot</summary>

tet___Veterans_Affairs


</details>


For a followup ticket probably, some backend work, up to you if you want to spin it out (and do it yourself, or we can ask @ethanteague). 

- [ ] Hide preview button on node view (we have another issue somewhere that's similar for another content type)
- [ ] Field disabling by role for the Path field and (probably!) the Promo type field. Only content admins should be able to edito those.

**Comment by msbtterswrth** [comment]
@kevwalsh I think I got everything. I decided to go with using the title field for the link text, so I updated that accordingly, I need to take a look later at why the behat tests are failing though.

File: config/sync/node.type.banner.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ third_party_settings:
   menu_force:
     menu_force: 0
     menu_force_parent: 0
-name: Banner
+name: 'Full Width Alert'
 type: banner
 description: 'A full width dismissible banner'
 help: ''
```
Comment (kevwalsh) [inline config/sync/node.type.banner.yml:21]: For Full Width Alerts

- [ ] 1. Move Alert type help text to Submission guidelines.
- [ ] 2. Change "Currently the CMS supports two types of alerts" to "www.VA.gov supports to types of Full Width Alerts.
- [ ] 3. Append to submission guidelines:`<p>Read more about Full Width Alerts in the <a href="/node/34828">Knowledge Base</a> (opens in a new window).</p>`

<details><summary>Screenshot</summary>
Create_Banner___VA_gov_CMS

</details>

File: config/sync/core.base_field_override.node.promo_banner.title.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+langcode: en
+status: true
+dependencies:
+  config:
+    - node.type.promo_banner
+id: node.promo_banner.title
+field_name: title
+entity_type: node
+bundle: promo_banner
+label: 'Name of promo banner'
```
Comment (kevwalsh) [inline config/sync/core.base_field_override.node.promo_banner.title.yml:11]: Change to "Link text"

File: config/sync/node.type.promo_banner.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  module:
+    - menu_force
+    - menu_ui
+    - node_title_help_text
+third_party_settings:
+  menu_ui:
+    available_menus: {  }
+    parent: ''
+  node_title_help_text:
+    title_help: 'Please enter the banner link text .'
```
Comment (kevwalsh) [inline config/sync/node.type.promo_banner.yml:14]: This is redundant, i think you can remove it.

File: config/sync/node.type.promo_banner.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  menu_force:
+    menu_force: 0
+    menu_force_parent: 0
+name: 'Promo Banner'
+type: promo_banner
+description: 'Promo banners are fixed content used for dismissible announcements such as new tools, news, etc.'
+help: ''
```
Comment (kevwalsh) [inline config/sync/node.type.promo_banner.yml:21]: For **Full Width Alerts**

- [ ] 1. Move Promo type help text to Submission guidelines.
- [ ] 2. Append to submission guidelines:`<p>Read more about Promo Banners in the <a href="/node/39828">Knowledge Base</a> (opens in a new window).</p>`
- [ ] 3. Change the label for "Name of promo" banner to "Link text" and remove help text. 

<details><summary>Screenshot</summary>

Cursor_and_Create_Promo_Banner___VA_gov_CMS


</details>

**Review by kevwalsh — CHANGES_REQUESTED** [review]
A few more fiddly things.

**Comment by kevwalsh** [comment]
@msbtterswrth one other update, could go in a separate issue, it should be possible for content admins to create these.  It's a small permissions change. Would you prefer to take care of that in this PR or in another one? 

<details><summary>Screenshot</summary>


Cursor_and_Permissions___VA_gov_CMS

</details>

**Comment by msbtterswrth** [comment]
@kevwalsh i went ahead and updated those perms in this PR, was jsut an oversight on my part so let's get it in. I also noticed that my behat tests were failing because the promo banner content type doesn't exist on the pr env, but shouldn't it be created on drush cim?

**Comment by kevwalsh** [comment]
> @kevwalsh. I also noticed that my behat tests were failing because the promo banner content type doesn't exist on the pr env, but shouldn't it be created on drush cim?

The content type is definitely there! i just checked. so i don't know. Tests are still running now.

**Review by kevwalsh — CHANGES_REQUESTED** [review]
One more thing, sorry. 

Content Editors, Reviewers, and Publishers all need 

- [ ] 1. Edit any Promo Banners (what "any" means is determined via Workbench Access  / Sections)
- [ ] 2. View Promo Banner revisions


Publisher needs
1. Revert Promo Banner revisions

Also i realized only when going back in to the permissions page that Content Admin didn't actually need this, they have "bypass node permissions". My bad! It's been awhile since i've spent time in node permissions. i think i'll create a node permissions cleanup issue separately,so you don't need to change this back, unless you're inspired to :) 

Also, per @swirtSJW 
- [ ] Remove default * for path.

File: config/sync/field.field.node.promo_banner.field_promo_type.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  config:
+    - field.storage.node.field_promo_type
+    - node.type.promo_banner
+  module:
+    - options
+id: node.promo_banner.field_promo_type
+field_name: field_promo_type
+entity_type: node
+bundle: promo_banner
+label: 'Promo Type'
```
Comment (swirtSJW) [inline config/sync/field.field.node.promo_banner.field_promo_type.yml:14]: Sentence case?

File: config/sync/field.field.node.promo_banner.field_target_paths.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+field_name: field_target_paths
+entity_type: node
+bundle: promo_banner
+label: Paths
+description: ''
+required: false
+translatable: true
+default_value:
+  -
+    value: '*'
```
Comment (swirtSJW) [inline config/sync/field.field.node.promo_banner.field_target_paths.yml:18]: No default.  https://dsva.slack.com/archives/CT4GZBM8F/p1638212935348900   Will allow it to use the same entity validation we do for banner.

File: config/sync/field.storage.node.field_promo_type.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+entity_type: node
+type: list_string
+settings:
+  allowed_values:
+    -
+      value: announcements
+      label: Announcements
+    -
+      value: email-signup
+      label: 'Email Signup'
```
Comment (swirtSJW) [inline config/sync/field.storage.node.field_promo_type.yml:19]: Sentence case?

File: config/sync/field.storage.node.field_promo_type.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    -
+      value: news-stories
+      label: 'News Stories'
```
Comment (swirtSJW) [inline config/sync/field.storage.node.field_promo_type.yml:22]: Sentence case?

File: config/sync/node.type.banner.yml [hunk 9]
```diff
@@ -15,10 +15,10 @@ third_party_settings:
   menu_force:
     menu_force: 0
     menu_force_parent: 0
-name: Banner
+name: 'Full Width Alert'
 type: banner
 description: 'A full width dismissible banner'
-help: ''
+help: "<p>Currently, www.va.gov supports to types of Full Width Alerts:</p>\r\n\r\n<ol>\r\n\t<li><strong>Informational alerts</strong>, which are used to provide helpful information to a user or something that warrants attention. Not used for negative consequences.</li>\r\n\t<li><strong>Warning alerts</strong>, which are used to warn a user, such as when there are negative consequences, but necessary when something has gone wrong.</li>\r\n</ol>\r\n<p>You can read more about these alert types in the <a href=\"https://design.va.gov/components/alertboxes\">VA Design System</a> or read more about Full Width Alerts in the <a href=\"/node/34828\" target=\"_blank\">Knowledge Base</a> (opens in a new window).</p>"
```
Comment (swirtSJW) [inline config/sync/node.type.banner.yml:21]:
```suggestion
help: "<p>Currently, www.va.gov supports two types of Full Width Alerts:</p>\r\n\r\n<ol>\r\n\t<li><strong>Informational alerts</strong>, which are used to provide helpful information to a user or something that warrants attention. Not used for negative consequences.</li>\r\n\t<li><strong>Warning alerts</strong>, which are used to warn a user, such as when there are negative consequences, but necessary when something has gone wrong.</li>\r\n</ol>\r\n<p>You can read more about these alert types in the <a href=\"https://design.va.gov/components/alertboxes\">VA Design System</a> or read more about Full Width Alerts in the <a href=\"/node/34828\" target=\"_blank\">Knowledge Base</a> (opens in a new window).</p>"
```

File: tests/behat/drupal-spec-tool/content_model_bundles.feature [hunk 10]
```diff
@@ -8,7 +8,7 @@ Feature: Content model bundles
      Scenario: Bundles
        Then exactly the following content entity type bundles should exist
        | Name | Machine name | Type | Description |
-| Banner | banner | Content type | A full width dismissible banner |
+| Full Width Alert | banner | Content type | A full width dismissible banner |
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_bundles.feature:11]: Re-alpha this

**Review by swirtSJW — CHANGES_REQUESTED** [review]
A few minor things.   Some I am not sure of.

File: config/sync/node.type.promo_banner.yml [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    parent: ''
+  node_title_help_text:
+    title_help: ''
+  menu_force:
+    menu_force: 0
+    menu_force_parent: 0
+name: 'Promo Banner'
+type: promo_banner
+description: 'Promo banners are fixed content used for dismissible announcements such as new tools, news, etc.'
+help: "<p>Currently, www.va.gov supports three types of promo banners:</p>\r\n\r\n<ol>\r\n\t<li><strong>Benefit announcements</strong>, which relate to content focused on benefits: a new tool, a new online benefit application, etc.</li>\r\n\t<li><strong>Email Signups</strong>, which relate to email signups for various VA updates / alerts / communications.</li>\r\n\t<li><strong>News Stories</strong>, which relate to newly received or noteworthy information about the VA.</li>\r\n</ol>\r\n\r\n<p>You can read more about these promo types in the <a href=\"https://design.va.gov/components/promo-banners\" target=\"_blank\">VA Design System</a> or read more about Promo Banners in the <a href=\"/node/39828\">Knowledge Base</a> (opens in a new window).</p>"
```
Comment (swirtSJW) [inline config/sync/node.type.promo_banner.yml:21]:
```suggestion
help: "<p>Currently, www.va.gov supports three types of promo banners:</p>\r\n\r\n<ol>\r\n\t<li><strong>Benefit announcements</strong>, which relate to content focused on benefits: a new tool, a new online benefit application, etc.</li>\r\n\t<li><strong>Email signups</strong>, which relate to email signups for various VA updates / alerts / communications.</li>\r\n\t<li><strong>News stories</strong>, which relate to newly received or noteworthy information about the VA.</li>\r\n</ol>\r\n\r\n<p>You can read more about these promo types in the <a href=\"https://design.va.gov/components/promo-banners\" target=\"_blank\">VA Design System</a> or read more about Promo Banners in the <a href=\"/node/39828\">Knowledge Base</a> (opens in a new window).</p>"
```

File: config/sync/core.entity_form_display.node.promo_banner.default.yml [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+        weight: 0
+        required_fields: true
+        id: ''
+        classes: ''
+        show_empty_fields: false
+      label: 'Section Settings'
+    group_editorial_workflow:
+      children:
+        - revision_log
+        - moderation_state
```
Comment (ethanteague) [inline config/sync/core.entity_form_display.node.promo_banner.default.yml:38]:
```suggestion
        - moderation_state
        - revision_log
```

Comment (ethanteague) [reply config/sync/core.entity_form_display.node.promo_banner.default.yml:38]: The revision_log test is expecting moderation_state to come before revision_log. Committing this suggestion should fix your failing test, @msbtterswrth

Comment (msbtterswrth) [reply config/sync/core.entity_form_display.node.promo_banner.default.yml]: You're amazing. thank you so much!

File: tests/behat/drupal-spec-tool/content_model_bundles.feature [hunk 13]
```diff
@@ -115,4 +116,3 @@ Feature: Content model bundles
 | Topics | topics | Vocabulary |  |
 | Type of Redirect | type_of_redirect | Vocabulary |  |
 | VHA health service taxonomy | health_care_service_taxonomy | Vocabulary | Single source of truth for health service names, descriptions, patient-friendly names, and common conditions. |
-
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/content_model_bundles.feature:118]: Minor:  @msbtterswrth  Please adjust your IDE to add a empty line to the end of the file if there isn't one for .feature files.  Its a drupal standard we apply to all files.

**Review by swirtSJW — APPROVED** [review]
This tests out OK.   But there is one issue, I think related to the governance and behavior of the paths.   On full width banner alerts, normal editors aren't able to edit the paths, but on this one they still can.   I think we can add that later though so merging this.

**Pull request merged**